src/presentations/Cart.js

```diff
@@ -0,0 +1,142 @@
+import React from 'react';
+import {connect} from 'react-redux';
+import LazyLoad from 'react-lazyload';
+import {resetCartData} from '../actions/productActions';
+class Cart extends React.Component{
+    constructor(props){
+        super(props);
+        this.state={
+            itemList: [],
+            totalCartValue:0
+        }
+        // this.increaseQty = this.increaseQty.bind(this);
+        // this.decreaseQty = this.decreaseQty.bind(this);
+        // this.evaluateTotalPrice = this.evaluateTotalPrice.bind(this);
+    }
+    componentDidMount(){
+        if(this.props.cartData){
+            this.setState({itemList:this.props.cartData},()=>{
+                this.evaluateTotalPrice();
+            });
+        }
+    }
+    increaseQty=(e,itm,i)=>{
+        let tempObj = Object.assign({},this.state);
+        itm.qty = itm.qty + 1;
+        tempObj.itemList[i].qty = itm.qty;
+        this.setState(tempObj,()=>{
+            this.evaluateTotalPrice();
+        });
+    }
+    decreaseQty=(e,itm,i)=>{
+        let tempObj = Object.assign({},this.state);
+        itm.qty = itm.qty - 1;
+        if(itm.qty == 0){
+            tempObj.itemList.splice(i,1);
+        }
+        if(!tempObj.itemList.length){
+            this.props.resetCartData();
+        }
+        this.setState(tempObj,()=>{
+            this.evaluateTotalPrice();
+        });
+    }
+    evaluateTotalPrice=()=>{
+        let itmLst = this.state.itemList;
+        let totalValue = 0;
+        // for(let i=0;i<itmLst.length;i++){
+        //     totalValue += itmLst[i].finalPrice;
+        // }
+        itmLst.forEach((item)=>{
+            totalValue+= item.finalPrice;
+        });
+        this.setState({totalCartValue: totalValue});
+    }
+    render(){
+        const {itemList,totalCartValue} = this.state;
+        const {hideHeader,closeModal,checkoutComplete} = this.props;
+        return(<main id="cart" role="dialog" tabIndex="10" aria-labelledby="dialog-title" className="cart-page">
+            {!hideHeader ? (<section className="cart-header modal-horizontal-space">
+                <h2 id="dialogue-title">My cart {itemList.length ? " ("+(itemList.length)+" items)" : " "}</h2>
```

Comment:
you have main on cart, main on home/product page, main on modal. You will end up having too many main on a single page.